Zig's formatting doesn't work using `editor: format` command

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Using `editor: format` on zig files does not format them, neither manually nor on save.
Trying vscode or using zig format manually works though.


### LSP logs:
```
Starting ZLS      0.14.0-dev.304+b401aab @ '/home/kofta/.local/bin/zls'
Log File:         /home/kofta/.cache/zls/zls.log (info)
Client Info:      Zed-0.166.1
Autofix Mode:     source.fixall
Offset Encoding:  utf-16
added Workspace Folder: file:///home/kofta/shit/zig-fmt
Set config option 'builtin_path' to "/home/kofta/.cache/zls/builtin.zig"
Set config option 'zig_lib_path' to "/home/kofta/Downloads/zig-linux-x86_64-0.14.0-dev.2557+f06ca14cb/lib"
Set config option 'zig_exe_path' to "/home/kofta/.local/bin/zig"
Set config option 'build_runner_path' to "/home/kofta/.cache/zls/build_runner/31755aaa5daba3be3b3e2b608dc8531e/build_runner.zig"
Set config option 'global_cache_path' to "/home/kofta/.cache/zls"
Took 7ms to process request-0-initialize on Thread 41825
Dynamically registering method 'workspace/didChangeConfiguration'
Took 0ms to process notification-initialized on Thread 41825
Took 0ms to process notification-workspace/didChangeConfiguration on Thread 41825
Loaded build file 'file:///home/kofta/shit/zig-fmt/build.zig'
Resolved build file of 'file:///home/kofta/shit/zig-fmt/src/main.zig' as 'file:///home/kofta/shit/zig-fmt/build.zig'
Opened document 'file:///home/kofta/shit/zig-fmt/src/main.zig'
Took 0ms to process notification-textDocument/didOpen on Thread 41825
Took 0ms to process response-"register-workspace/didChangeConfiguration" on Thread 41825
trying to start Build-On-Save for 'file:///home/kofta/shit/zig-fmt'
Took 3ms to process response-"i_haz_configuration" on Thread 41825
Took 3ms to process response-"i_haz_configuration" on Thread 41825
zig build runner process has exited
Took 3ms to process request-2-textDocument/codeAction on Thread 41828
Took 4ms to process request-3-textDocument/codeAction on Thread 41829
Took 3ms to process request-4-textDocument/codeAction on Thread 41827
Took 3ms to process request-5-textDocument/codeAction on Thread 41826
Took 3ms to process request-6-textDocument/codeAction on Thread 41828
Took 3ms to process request-7-textDocument/codeAction on Thread 41829
Took 4ms to process request-8-textDocument/codeAction on Thread 41827
```

### Zed config:
```jsonc
{
  "vim_mode": true,
  "languages": {
    "Zig": {
      // Formatting with ZLS matches `zig fmt`.
      // The Zig FAQ answers some questions about `zig fmt`:
      // https://github.com/ziglang/zig/wiki/FAQ
      "format_on_save": "language_server",
      "code_actions_on_format": {
        // Run code actions that currently supports adding and removing discards.
        // "source.fixAll": true,
        // Run code actions that sorts @import declarations.
        // Available since ZLS `0.14.0-dev.188+2be424de5`
        // "source.organizeImports": true,
      }
    }
  },
  "lsp": {
    "zls": {
      "binary": {
        // omit the following line if `zls` is in your PATH
        "path": "/usr/local/bin/zls"
      },
      // There are two ways to set config options:
      //   - edit your `zls.json` that applies to any editor that uses ZLS
      //   - set in-editor config options with the `initialization_options` field below.
      //
      // Further information on how to configure ZLS:
      // https://zigtools.org/zls/configure/
      //
      // Support for `initializationOptions` in Zed requires at least ZLS `0.14.0-dev.22+a263b8dc6`.
      "initialization_options": {
        // Whether to enable build-on-save diagnostics
        //
        // Further information about build-on save:
        // https://zigtools.org/zls/guides/build-on-save/
        // "enable_build_on_save": true,

        // omit the following line if `zig` is in your PATH
        "zig_exe_path": "/usr/local/bin/zig"
      }
    }
  }
}
```


### Zed Version and System Specs

Zed: v0.166.1 (Zed)
OS: Linux Wayland fedora 41
Memory: 15.4 GiB
Architecture: x86_64
GPU: Intel(R) UHD Graphics (CML GT2) || Intel open-source Mesa driver || Mesa 24.2.7

### If applicable, add screenshots or screencasts of the incorrect state / behavior

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

<!-- Click below this line and paste or drag-and-drop your log-->
```
2024-12-25T11:03:00.653278199+02:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "Intel(R) UHD Graphics (CML GT2)", driver_name: "Intel open-source Mesa driver", driver_info: "Mesa 24.2.7" }
2024-12-25T11:03:00.655963465+02:00 [INFO] Using surface present mode MAILBOX
2024-12-25T11:03:00.656312253+02:00 [WARN] Unable to forbid exclusive full screen
2024-12-25T11:03:00.656494549+02:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-12-25T11:03:00.692037309+02:00 [INFO] Using surface present mode MAILBOX
2024-12-25T11:03:00.692431798+02:00 [WARN] Unable to forbid exclusive full screen
2024-12-25T11:03:00.693268706+02:00 [INFO] Using surface present mode MAILBOX
2024-12-25T11:03:00.693687465+02:00 [WARN] Unable to forbid exclusive full screen
2024-12-25T11:03:00.778209771+02:00 [INFO] Language server with id 3 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:03:00.798898553+02:00 [INFO] using project environment variables from CLI. PATH="/home/kofta/.nvm/versions/node/v23.2.0/bin:/home/kofta/.local/share/pnpm:/home/kofta/.deno/bin:/home/kofta/.fly/bin:/home/kofta/.local/bin:/home/kofta/.asdf/shims:/home/kofta/.asdf/bin:/home/kofta/.bun/bin:/usr/lib64/ccache:/home/kofta/.cargo/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/home/kofta/.cache/rebar3/bin:/home/kofta/bin/"
2024-12-25T11:03:03.470187752+02:00 [INFO] attempting to start language server "discord_presence", path: "/home/kofta/shit/zig-fmt", id: 7
2024-12-25T11:03:03.470275561+02:00 [INFO] attempting to start language server "snippet-completion-server", path: "/home/kofta/shit/zig-fmt", id: 8
2024-12-25T11:03:03.470327838+02:00 [INFO] attempting to start language server "zls", path: "/home/kofta/shit/zig-fmt", id: 9
2024-12-25T11:03:03.47187671+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/share/zed/extensions/work/snippets/simple-completion-language-server-v0.1.1/simple-completion-language-server", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:03:03.472232105+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/share/zed/extensions/work/discord-presence/discord-presence-lsp-v0.6.0/discord-presence-lsp-x86_64-unknown-linux-gnu/discord-presence-lsp", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:03:03.472725197+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/bin/zls", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:03:03.490071331+02:00 [WARN] unhandled capability registration: Registration { id: "register-workspace/didChangeConfiguration", method: "workspace/didChangeConfiguration", register_options: None }
2024-12-25T11:03:03.495525227+02:00 [INFO] Language server with id 9 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:03:03.507438136+02:00 [INFO] Language server with id 9 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:03:03.581751967+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:05.571059216+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/df44767c7c150253104050bdb20fe270" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:05.571251143+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/tmp/87290c898207e702" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:06.349787471+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/df44767c7c150253104050bdb20fe270/build.o" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:10.202809146+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/df44767c7c150253104050bdb20fe270/build.o" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:10.465839563+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/1b174355d28ca50e84264ff52515c1c6.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:10.465970527+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/timestamp" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:10.466099119+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/df44767c7c150253104050bdb20fe270/build" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:10.466229689+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/df44767c7c150253104050bdb20fe270/build.tmp34e6da3" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:10.466408238+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/df44767c7c150253104050bdb20fe270/build.tmp783c94c" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:23.976018606+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/1b174355d28ca50e84264ff52515c1c6.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:23.976226787+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/timestamp" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:23.97642399+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/tmp/f6475e13ade38ac8" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:37.28219748+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/416710afefa6ba9b7ea6e0447401f0bb.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:37.282376733+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/timestamp" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:37.282576809+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/tmp/281fb3b141e2a947" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:37.28273276+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/tmp/4ed145cc2091349b" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:37.282886247+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/z/954fa778e285a9cbdb0b85c7239cfda9" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:39.461846032+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/366033f59f4b5174417dc3cb0cab70bb" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:39.46203879+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/tmp/281fb3b141e2a947" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:40.298421674+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/366033f59f4b5174417dc3cb0cab70bb/build.o" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.553647815+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/366033f59f4b5174417dc3cb0cab70bb/build.o" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.839830128+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/416710afefa6ba9b7ea6e0447401f0bb.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.839999698+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/945e67dbd89f1f819f2f15a41255170a.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.840117012+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/fdb1e119c835d2beb1da69f95f79a51f.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.840237218+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/timestamp" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.840356559+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/366033f59f4b5174417dc3cb0cab70bb/build" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.840478209+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/366033f59f4b5174417dc3cb0cab70bb/build.tmp810a0da" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.840591864+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/366033f59f4b5174417dc3cb0cab70bb/build.tmpc6b47e0" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.840701044+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/tmp/271e40617fa1be94" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.840935213+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/tmp/45e41d87ddfb9a54" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.841094041+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/z/213979b7a051c3219186b58a88982b5e" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:44.841251315+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/z/4c01f0fc717bf031e4d26ed9292e126a" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:45.269979034+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/07af9f2e866812db1d4c98168b5dd371" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:45.270170421+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/6ad37384ed1c37989572b6ac015c8007" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:45.270339692+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/tmp/271e40617fa1be94" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:45.27047006+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/tmp/45e41d87ddfb9a54" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:45.439868056+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/07af9f2e866812db1d4c98168b5dd371/libzig-fmt.a.o" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:45.440059518+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/6ad37384ed1c37989572b6ac015c8007/zig-fmt.o" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.284262831+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/945e67dbd89f1f819f2f15a41255170a.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.284451869+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/07af9f2e866812db1d4c98168b5dd371/libzig-fmt.a" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.284620756+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/07af9f2e866812db1d4c98168b5dd371/libzig-fmt.a.o" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.284877542+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/07af9f2e866812db1d4c98168b5dd371/libzig-fmt.a.temp-archive-573c8ce.a" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.285027842+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/6ad37384ed1c37989572b6ac015c8007/zig-fmt.o" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.285143053+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/zig-out" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.386672696+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/fdb1e119c835d2beb1da69f95f79a51f.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.386815919+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/6ad37384ed1c37989572b6ac015c8007/zig-fmt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.386936081+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/6ad37384ed1c37989572b6ac015c8007/zig-fmt.tmp010a033" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.3870669+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/o/6ad37384ed1c37989572b6ac015c8007/zig-fmt.tmp72f4e65" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:46.387182577+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/zig-out/bin" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:50.150356464+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/416710afefa6ba9b7ea6e0447401f0bb.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:50.150509076+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/945e67dbd89f1f819f2f15a41255170a.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:50.150656777+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/h/fdb1e119c835d2beb1da69f95f79a51f.txt" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:03:50.150784639+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/.zig-cache/tmp/4c7e63b1424d59c7" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:04:15.84654355+02:00 [ERROR] ignoring event "/home/kofta/shit/zig-fmt/src/main.zig" outside of root path "/home/kofta/shit/zig-testing"
2024-12-25T11:05:03.473438999+02:00 [ERROR] Cancelled LSP request task for "initialize" id 0 which took over 120s
2024-12-25T11:05:03.473586246+02:00 [ERROR] oneshot canceled
2024-12-25T11:05:03.473641824+02:00 [ERROR] Failed to start language server "discord_presence": LSP request timeout
2024-12-25T11:05:03.473689563+02:00 [ERROR] server stderr: "thread 'main' panicked at lsp/src/discord.rs:61:16:\ncalled `Result::unwrap()` on an `Err` value: \"Couldn't connect to the Discord IPC socket\"\nnote: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\n"
2024-12-25T11:05:53.721697614+02:00 [ERROR] oneshot canceled
2024-12-25T11:05:53.730061052+02:00 [WARN] skipping diagnostics update, no worktree found for path "/home/kofta/.config/zed/settings.json"
2024-12-25T11:05:55.742215123+02:00 [INFO] ========== starting zed ==========
2024-12-25T11:05:55.771219809+02:00 [INFO] perform;
2024-12-25T11:05:55.771312208+02:00 [INFO] authenticate;
2024-12-25T11:05:55.771385082+02:00 [INFO] Opening main db
2024-12-25T11:05:55.771457168+02:00 [INFO] read_command;
2024-12-25T11:05:55.771552799+02:00 [INFO] send_secondary_commands;
2024-12-25T11:05:55.77186614+02:00 [INFO] new;
2024-12-25T11:05:55.771946781+02:00 [INFO] socket reader;
2024-12-25T11:05:55.772295024+02:00 [INFO] Opening main db
2024-12-25T11:05:55.772610888+02:00 [INFO] keep_updated;
2024-12-25T11:06:00.149767367+02:00 [ERROR] oneshot canceled
2024-12-25T11:06:00.149874175+02:00 [ERROR] oneshot canceled
2024-12-25T11:06:00.149976056+02:00 [ERROR] oneshot canceled
2024-12-25T11:06:00.150049688+02:00 [ERROR] oneshot canceled
2024-12-25T11:06:00.156190498+02:00 [ERROR] oneshot canceled
2024-12-25T11:06:01.477180007+02:00 [INFO] ========== starting zed ==========
2024-12-25T11:06:01.515716834+02:00 [INFO] perform;
2024-12-25T11:06:01.51578216+02:00 [INFO] authenticate;
2024-12-25T11:06:01.51584228+02:00 [INFO] read_command;
2024-12-25T11:06:01.515899602+02:00 [INFO] Opening main db
2024-12-25T11:06:01.515951677+02:00 [INFO] send_secondary_commands;
2024-12-25T11:06:01.516220164+02:00 [INFO] new;
2024-12-25T11:06:01.51627292+02:00 [INFO] socket reader;
2024-12-25T11:06:01.516804283+02:00 [INFO] keep_updated;
2024-12-25T11:06:01.516851751+02:00 [INFO] Opening main db
2024-12-25T11:06:01.519341384+02:00 [INFO] Using git binary path: None
2024-12-25T11:06:01.562194358+02:00 [INFO] No prompt template overrides directory found at /home/kofta/.local/share/zed/prompt_overrides. Using built-in prompts.
2024-12-25T11:06:01.563040052+02:00 [INFO] extensions updated. loading 21, reloading 0, unloading 0
2024-12-25T11:06:01.565104036+02:00 [INFO] activate is not implemented on Linux, ignoring the call
2024-12-25T11:06:01.568379+02:00 [INFO] perform;
2024-12-25T11:06:01.568451496+02:00 [INFO] authenticate;
2024-12-25T11:06:01.568536766+02:00 [INFO] read_command;
2024-12-25T11:06:01.569076848+02:00 [INFO] socket reader;
2024-12-25T11:06:01.572586391+02:00 [WARN] Theme "One Dark Pro" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-25T11:06:01.576309155+02:00 [INFO] new;
2024-12-25T11:06:01.576951582+02:00 [INFO] keep_updated;
2024-12-25T11:06:01.577365422+02:00 [WARN] Theme "One Dark Pro Test" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-25T11:06:01.578478699+02:00 [INFO] set status on client 0: Authenticating
2024-12-25T11:06:01.578776856+02:00 [INFO] perform;
2024-12-25T11:06:01.578840038+02:00 [INFO] authenticate;
2024-12-25T11:06:01.578896071+02:00 [INFO] read_command;
2024-12-25T11:06:01.579009454+02:00 [INFO] Opening main db
2024-12-25T11:06:01.57913783+02:00 [INFO] socket reader;
2024-12-25T11:06:01.581665034+02:00 [WARN] Theme "Smooth Dark" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-25T11:06:01.584578366+02:00 [WARN] Theme "The Dark Side" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-25T11:06:01.585817145+02:00 [INFO] new;
2024-12-25T11:06:01.586654343+02:00 [INFO] keep_updated;
2024-12-25T11:06:01.607242488+02:00 [INFO] Enabling Vulkan Portability
2024-12-25T11:06:01.607298238+02:00 [INFO] Enabling color space support
2024-12-25T11:06:01.735284736+02:00 [INFO] Adapter: "Intel(R) UHD Graphics (CML GT2)"
2024-12-25T11:06:01.735358651+02:00 [INFO] Testing presentation capability on Linux/Intel
2024-12-25T11:06:01.73710703+02:00 [INFO] No ray tracing extensions are supported
2024-12-25T11:06:01.738089316+02:00 [INFO] Using surface present mode MAILBOX
2024-12-25T11:06:01.738418763+02:00 [WARN] Unable to forbid exclusive full screen
2024-12-25T11:06:01.738582392+02:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-12-25T11:06:01.738696586+02:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-12-25T11:06:01.738817856+02:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: PreMultiplied }
2024-12-25T11:06:01.746341208+02:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "Intel(R) UHD Graphics (CML GT2)", driver_name: "Intel open-source Mesa driver", driver_info: "Mesa 24.2.7" }
2024-12-25T11:06:01.748466071+02:00 [INFO] Using surface present mode MAILBOX
2024-12-25T11:06:01.748743946+02:00 [WARN] Unable to forbid exclusive full screen
2024-12-25T11:06:01.74892309+02:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-12-25T11:06:01.766057711+02:00 [INFO] set status on client 154241: Connecting
2024-12-25T11:06:01.771170215+02:00 [INFO] Opening main db
2024-12-25T11:06:01.781382628+02:00 [INFO] Using surface present mode MAILBOX
2024-12-25T11:06:01.781892795+02:00 [WARN] Unable to forbid exclusive full screen
2024-12-25T11:06:01.78285805+02:00 [INFO] Using surface present mode MAILBOX
2024-12-25T11:06:01.783259066+02:00 [WARN] Unable to forbid exclusive full screen
2024-12-25T11:06:01.788141833+02:00 [ERROR] missing field `name` at line 199 column 1
2024-12-25T11:06:01.788292183+02:00 [INFO] Opening main db
2024-12-25T11:06:01.985043008+02:00 [INFO] using project environment variables from CLI. PATH="/home/kofta/.nvm/versions/node/v23.2.0/bin:/home/kofta/.local/share/pnpm:/home/kofta/.deno/bin:/home/kofta/.fly/bin:/home/kofta/.local/bin:/home/kofta/.asdf/shims:/home/kofta/.asdf/bin:/home/kofta/.bun/bin:/usr/lib64/ccache:/home/kofta/.cargo/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/home/kofta/.cache/rebar3/bin:/home/kofta/bin/"
2024-12-25T11:06:02.056175297+02:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-12-25T11:06:02.060959786+02:00 [INFO] starting language server process. binary path: "/home/kofta/.nvm/versions/node/v23.2.0/bin/node", working directory: "/", args: ["/home/kofta/.local/share/zed/copilot/copilot-v0.7.0/dist/language-server.js", "--stdio"]
2024-12-25T11:06:02.192668765+02:00 [ERROR] cannot read LSP message headers
2024-12-25T11:06:02.192803357+02:00 [ERROR] Broken pipe (os error 32)
2024-12-25T11:06:02.192868196+02:00 [ERROR] server shut down
2024-12-25T11:06:02.568655594+02:00 [INFO] add_connection;
2024-12-25T11:06:02.573429493+02:00 [INFO] set environment variables from shell:/usr/bin/zsh, path:/home/kofta/.deno/bin:/home/kofta/.fly/bin:/home/kofta/.local/bin:/home/kofta/.bun/bin:/home/kofta/.nvm/versions/node/v23.2.0/bin:/home/kofta/.local/share/pnpm:/home/kofta/.deno/bin:/home/kofta/.fly/bin:/home/kofta/.local/bin:/home/kofta/.asdf/shims:/home/kofta/.asdf/bin:/home/kofta/.bun/bin:/usr/lib64/ccache:/home/kofta/.cargo/bin:/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/home/kofta/.cache/rebar3/bin:/home/kofta/bin/:/home/kofta/.cache/rebar3/bin:/home/kofta/bin/
2024-12-25T11:06:02.581271127+02:00 [INFO] set status on client 154241: Connected { peer_id: PeerId { owner_id: 629, id: 5501753 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-12-25T11:06:03.986311998+02:00 [INFO] attempting to start language server "discord_presence", path: "/home/kofta/shit/zig-fmt", id: 1
2024-12-25T11:06:03.986394067+02:00 [INFO] attempting to start language server "snippet-completion-server", path: "/home/kofta/shit/zig-fmt", id: 2
2024-12-25T11:06:03.986451922+02:00 [INFO] attempting to start language server "zls", path: "/home/kofta/shit/zig-fmt", id: 3
2024-12-25T11:06:03.987212288+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/bin/zls", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:06:03.999291961+02:00 [WARN] unhandled capability registration: Registration { id: "register-workspace/didChangeConfiguration", method: "workspace/didChangeConfiguration", register_options: None }
2024-12-25T11:06:04.006677258+02:00 [INFO] Language server with id 3 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:06:04.014094147+02:00 [INFO] Language server with id 3 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:06:04.25274327+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/share/zed/extensions/work/snippets/simple-completion-language-server-v0.1.1/simple-completion-language-server", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:06:04.383788693+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/share/zed/extensions/work/discord-presence/discord-presence-lsp-v0.6.0/discord-presence-lsp-x86_64-unknown-linux-gnu/discord-presence-lsp", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:08:04.384544035+02:00 [ERROR] Cancelled LSP request task for "initialize" id 0 which took over 120s
2024-12-25T11:08:04.385040522+02:00 [ERROR] oneshot canceled
2024-12-25T11:08:04.385294853+02:00 [ERROR] Failed to start language server "discord_presence": LSP request timeout
2024-12-25T11:08:04.385623724+02:00 [ERROR] server stderr: "thread 'main' panicked at lsp/src/discord.rs:61:16:\ncalled `Result::unwrap()` on an `Err` value: \"Couldn't connect to the Discord IPC socket\"\nnote: run with `RUST_BACKTRACE=1` environment variable to display a backtrace\n"
2024-12-25T11:11:00.953753145+02:00 [INFO] Using surface present mode MAILBOX
2024-12-25T11:11:00.95413975+02:00 [WARN] Unable to forbid exclusive full screen
2024-12-25T11:11:01.889555043+02:00 [INFO] Using surface present mode MAILBOX
2024-12-25T11:11:01.889843606+02:00 [WARN] Unable to forbid exclusive full screen
2024-12-25T11:11:19.22063536+02:00 [INFO] open paths ["/home/kofta/.config/zed/settings.json"]
2024-12-25T11:11:19.226332755+02:00 [INFO] attempting to start language server "json-language-server", path: "/home/kofta/.config/zed/settings.json", id: 4
2024-12-25T11:11:19.226413506+02:00 [INFO] attempting to start language server "discord_presence", path: "/home/kofta/.config/zed/settings.json", id: 5
2024-12-25T11:11:19.226472826+02:00 [INFO] Initializing default prettier with plugins {}
2024-12-25T11:11:19.226678619+02:00 [INFO] Initializing default prettier with plugins {}
2024-12-25T11:11:19.227478036+02:00 [INFO] fetching latest version of language server "json-language-server"
2024-12-25T11:11:19.228245463+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/share/zed/extensions/work/discord-presence/discord-presence-lsp-v0.6.0/discord-presence-lsp-x86_64-unknown-linux-gnu/discord-presence-lsp", working directory: "/home/kofta/.config/zed", args: []
2024-12-25T11:11:19.241484416+02:00 [INFO] Language server with id 3 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:11:19.979161879+02:00 [INFO] downloading language server "json-language-server"
2024-12-25T11:11:20.189688377+02:00 [INFO] starting language server process. binary path: "/home/kofta/.nvm/versions/node/v23.2.0/bin/node", working directory: "/home/kofta/.config/zed", args: ["/home/kofta/.local/share/zed/languages/json-language-server/node_modules/vscode-langservers-extracted/bin/vscode-json-language-server", "--stdio"]
2024-12-25T11:11:22.406565649+02:00 [ERROR] EOF while parsing a value at line 200 column 0
2024-12-25T11:11:22.408916779+02:00 [ERROR] Failed to load user settings: EOF while parsing a value at line 200 column 0
2024-12-25T11:11:22.409198356+02:00 [INFO] Initializing default prettier with plugins {}
2024-12-25T11:11:22.428727387+02:00 [INFO] Language server with id 3 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:11:24.270804148+02:00 [ERROR] EOF while parsing a value at line 201 column 0
2024-12-25T11:11:24.270906933+02:00 [ERROR] Failed to load user settings: EOF while parsing a value at line 201 column 0
2024-12-25T11:11:24.271216597+02:00 [INFO] Initializing default prettier with plugins {}
2024-12-25T11:11:24.290203272+02:00 [INFO] Language server with id 3 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:11:25.965917479+02:00 [ERROR] Failed to load user settings: EOF while parsing a value at line 204 column 0
2024-12-25T11:11:25.966214727+02:00 [INFO] Initializing default prettier with plugins {}
2024-12-25T11:11:25.972225457+02:00 [ERROR] EOF while parsing a value at line 204 column 0
2024-12-25T11:11:25.990237506+02:00 [INFO] Language server with id 3 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:11:26.984012561+02:00 [ERROR] key must be a string at line 23 column 7
2024-12-25T11:11:26.990370677+02:00 [INFO] Initializing default prettier with plugins {}
2024-12-25T11:11:26.990441435+02:00 [INFO] stopping language server discord_presence
2024-12-25T11:11:26.990503302+02:00 [INFO] stopping language server snippet-completion-server
2024-12-25T11:11:26.990546032+02:00 [INFO] stopping language server zls
2024-12-25T11:11:27.020217423+02:00 [ERROR] oneshot canceled
2024-12-25T11:11:27.020303839+02:00 [ERROR] oneshot canceled
2024-12-25T11:11:27.020424504+02:00 [ERROR] cannot read LSP message headers
2024-12-25T11:11:27.020651084+02:00 [INFO] attempting to start language server "discord_presence", path: "/home/kofta/shit/zig-fmt", id: 6
2024-12-25T11:11:27.020718165+02:00 [INFO] attempting to start language server "snippet-completion-server", path: "/home/kofta/shit/zig-fmt", id: 7
2024-12-25T11:11:27.020781987+02:00 [INFO] attempting to start language server "zls", path: "/home/kofta/shit/zig-fmt", id: 8
2024-12-25T11:11:27.021037398+02:00 [INFO] starting language server process. binary path: "/path/to/zls_executable", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:11:27.022087208+02:00 [ERROR] Failed to start language server "zls": failed to spawn command. path: "/path/to/zls_executable", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:11:27.022150844+02:00 [ERROR] server stderr: ""
2024-12-25T11:11:27.022266326+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/share/zed/extensions/work/snippets/simple-completion-language-server-v0.1.1/simple-completion-language-server", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:11:27.02254231+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/share/zed/extensions/work/discord-presence/discord-presence-lsp-v0.6.0/discord-presence-lsp-x86_64-unknown-linux-gnu/discord-presence-lsp", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:11:36.84683447+02:00 [INFO] Waiting for default prettier to install
2024-12-25T11:11:36.846901073+02:00 [INFO] Starting prettier at path "/home/kofta/.local/share/zed/prettier"
2024-12-25T11:11:36.944173986+02:00 [INFO] starting language server process. binary path: "/home/kofta/.nvm/versions/node/v23.2.0/bin/node", working directory: "/home/kofta/.local/share/zed/prettier", args: ["/home/kofta/.local/share/zed/prettier/prettier_server.js", "/home/kofta/.local/share/zed/prettier"]
2024-12-25T11:11:37.019118922+02:00 [INFO] Started default prettier in "/home/kofta/.local/share/zed/prettier"
2024-12-25T11:11:37.186617774+02:00 [ERROR] missing field `name` at line 44 column 1
2024-12-25T11:12:09.862004132+02:00 [ERROR] missing field `name` at line 44 column 1
2024-12-25T11:12:09.864955479+02:00 [INFO] stopping language server discord_presence
2024-12-25T11:12:09.865020562+02:00 [INFO] stopping language server snippet-completion-server
2024-12-25T11:12:09.865062771+02:00 [INFO] stopping language server zls
2024-12-25T11:12:09.875357275+02:00 [ERROR] oneshot canceled
2024-12-25T11:12:09.875469998+02:00 [ERROR] cannot read LSP message headers
2024-12-25T11:12:14.866329139+02:00 [INFO] timeout waiting for language server discord_presence to finish launching before stopping
2024-12-25T11:12:14.866424455+02:00 [INFO] attempting to start language server "discord_presence", path: "/home/kofta/shit/zig-fmt", id: 10
2024-12-25T11:12:14.866484672+02:00 [INFO] attempting to start language server "snippet-completion-server", path: "/home/kofta/shit/zig-fmt", id: 11
2024-12-25T11:12:14.866557235+02:00 [INFO] attempting to start language server "zls", path: "/home/kofta/shit/zig-fmt", id: 12
2024-12-25T11:12:14.866702194+02:00 [INFO] starting language server process. binary path: "/usr/local/bin/zls", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:12:14.866791754+02:00 [ERROR] oneshot canceled
2024-12-25T11:12:14.867367128+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/share/zed/extensions/work/discord-presence/discord-presence-lsp-v0.6.0/discord-presence-lsp-x86_64-unknown-linux-gnu/discord-presence-lsp", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:12:14.867657156+02:00 [INFO] starting language server process. binary path: "/home/kofta/.local/share/zed/extensions/work/snippets/simple-completion-language-server-v0.1.1/simple-completion-language-server", working directory: "/home/kofta/shit/zig-fmt", args: []
2024-12-25T11:12:14.867995492+02:00 [INFO] Language server with id 12 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "Starting ZLS      0.14.0-dev.304+b401aab @ '/usr/local/bin/zls'"
}
2024-12-25T11:12:14.885886115+02:00 [WARN] unhandled capability registration: Registration { id: "register-workspace/didChangeConfiguration", method: "workspace/didChangeConfiguration", register_options: None }
2024-12-25T11:12:14.894105075+02:00 [INFO] Language server with id 12 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:12:14.899510253+02:00 [INFO] Language server with id 12 sent unhandled notification workspace/semanticTokens/refresh:
null
2024-12-25T11:12:16.936388543+02:00 [ERROR] buffer has no file

```
<!-- Click above this line and paste or drag-and-drop your log--></details>

--- a/assets/settings/default.json
+++ b/assets/settings/default.json
@@ -1103,6 +1103,9 @@
       "prettier": {
         "allowed": true
       }
+    },
+    "Zig": {
+      "language_servers": ["zls", "..."]
     }
   },
   // Different settings for specific language models.

--- a/crates/lsp/src/lsp.rs
+++ b/crates/lsp/src/lsp.rs
@@ -284,6 +284,7 @@ impl<F: Future> LspRequestFuture<F::Output> for LspRequest<F> {
 }
 
 /// Combined capabilities of the server and the adapter.
+#[derive(Debug)]
 pub struct AdapterServerCapabilities {
     // Reported capabilities by the server
     pub server_capabilities: ServerCapabilities,

